backend/src/schema/resolvers/socialMedia.spec.js

```diff
@@ -1,186 +1,247 @@
+import { gql } from '../../jest/helpers'
+import { neode, getDriver } from '../../bootstrap/neo4j'
 
+const driver = getDriver()
 const factory = Factory()
+const instance = neode()
 
 describe('SocialMedia', () => {
-  let client, headers, variables, mutation
+  let socialMediaAction, someUser, ownerNode, owner
 
   const ownerParams = {
-    email: 'owner@example.com',
+    email: 'pippi@example.com',
     password: '1234',
-    id: '1234',
     name: 'Pippi Langstrumpf',
   }
 
   const userParams = {
-    email: 'someuser@example.com',
+    email: 'kalle@example.com',
     password: 'abcd',
-    id: 'abcd',
     name: 'Kalle Blomqvist',
   }
 
   const url = 'https://twitter.com/pippi-langstrumpf'
   const newUrl = 'https://twitter.com/bullerby'
 
-  const createSocialMediaMutation = gql`
-    mutation($url: String!) {
-      CreateSocialMedia(url: $url) {
-        id
-        url
-      }
-    }
-  `
-  const updateSocialMediaMutation = gql`
-    mutation($id: ID!, $url: String!) {
-      UpdateSocialMedia(id: $id, url: $url) {
-        id
-        url
-      }
-    }
-  `
-  const deleteSocialMediaMutation = gql`
-    mutation($id: ID!) {
-      DeleteSocialMedia(id: $id) {
-        id
-        url
-      }
-    }
-  `
+  const setUpSocialMedia = async () => {
+    const socialMediaNode = await instance.create('SocialMedia', { url })
+    await socialMediaNode.relateTo(ownerNode, 'ownedBy')
+    return socialMediaNode.toJson()
+  }
+
   beforeEach(async () => {
-    await factory.create('User', userParams)
-    await factory.create('User', ownerParams)
+    const someUserNode = await instance.create('User', userParams)
+    someUser = await someUserNode.toJson()
+    ownerNode = await instance.create('User', ownerParams)
+    owner = await ownerNode.toJson()
+
+    socialMediaAction = async (user, mutation, variables) => {
+      const { server } = createServer({
+        context: () => {
+          return {
+            user,
+            driver,
+          }
+        },
+      })
+      const { mutate } = createTestClient(server)
+
+      return mutate({
+        mutation,
+        variables,
+      })
+    }
   })
 
   afterEach(async () => {
     await factory.cleanDatabase()
   })
 
   describe('create social media', () => {
+    let mutation, variables
+
     beforeEach(() => {
+      mutation = gql`
+        mutation($url: String!) {
+          CreateSocialMedia(url: $url) {
+            id
+            url
+          }
+        }
+      `
       variables = { url }
-      mutation = createSocialMediaMutation
     })
 
     describe('unauthenticated', () => {
       it('throws authorization error', async () => {
-        client = new GraphQLClient(host)
-        await expect(client.request(mutation, variables)).rejects.toThrow('Not Authorised')
+        const user = null
+        const result = await socialMediaAction(user, mutation, variables)
+
+        expect(result.errors[0]).toHaveProperty('message', 'Not Authorised!')
```

Comment:
:heart_eyes: I learned where to look for the error message with `apollo-server-testing`.

Reply:
:smile: Better yet are the nested error messages...

By the way, I read some more about apollo error handling yesterday and I think the reason a request resolves with errors is to allow partial failures – e.g. when you request a `User` but for some reason can't fetch his `SocialMedia` you will still get a response with all the other user data, plus an error telling you (hopefully) which part of the request went wrong and why.